### DISCUSSION TITLE
GPT 学术优化 3.83 新申请的OpenAI API识别出错

### DISCUSSION THREAD

**Discussion by Suul042** [binary-husky/gpt_academic#1932] [Q&A] [answered]
所有对话记录将自动保存在本地目录gpt_log\admin\chat_secrets.log, 请注意自我隐私保护哦！
代理配置 无, 代理所在地：New Zealand
 [API_KEY] 本项目现已支持OpenAI和Azure的api-key。也支持同时填写多个api-key，如API_KEY="openai-key1,openai-key2,azure-key3"
 [API_KEY] 您既可以在config.py中修改api-key(s)，也可以在问题输入区输入临时的api-key(s)，然后回车键提交后即可生效。
 [API_KEY] 您的 API_KEY 不满足任何一种已知的密钥格式，请在config文件中修改API密钥之后再运行。

我的API_KEY是2024年8月10日生成的，但是添加之后，一直这样提示，在对话框输入也不可以，是不是对新的key识别不了呢？能否在界面上有一个api key的填写和验证呢？因为这个key我在别的软件是可以使用的，这是出现什么问题呢？

**Comment by linxi220520** [comment]
我的也是这个问题

**Reply by Suul042** [reply]
新版key应该是添加了下划线，所以识别不出来了，我查看老版本的key，没有下划线

**Reply by Suul042** [reply]
找到了一个解决方法，下载3.75版本，然后其他不变，依然按照原来的方式安装，文件夹下载最新的，然后，key验证就可以通过了，打开之后也是3.83，GPT 4o mini可以使用

**Comment by Yakefo** [comment, marked as answer]
我的是说缺少api_key。
临时解决方案：直接在输入区键入api_key，然后回车提交。
长效解决方案：在config.py中配置。

但是我确实已经填写了的  临时或者长效，都不可以 ，奇怪的很

**Reply by Suul042** [reply]
j就是不认新的格式的key，不知道为什么不发送验证，而是本地直接判断不能使用，因为我填在其他的里面都是可以使用的

**Reply by Suul042** [reply]
找到了一个解决方法，下载3.75版本，然后其他不变，依然按照原来的方式安装，文件夹下载最新的，然后，key验证就可以通过了，打开之后也是3.83，GPT 4o mini可以使用

**Reply by Suul042** [reply]
我回答的就是这个，下载3.75版本，然后其他不变，依然按照原来的方式安装，文件夹下载最新的，把key填在config.py里面就可以识别到了

**Reply by Yakefo** [reply]
> 我回答的就是这个，下载3.75版本，然后其他不变，依然按照原来的方式安装，文件夹下载最新的，把key填在config.py里面就可以识别到了

问个额外的问题： 免费的api key是不是用不成啊

**Comment by Suul042** [comment]
找到了一个解决方法，下载3.75版本，然后其他不变，依然按照原来的方式安装，文件夹下载最新的，然后，key验证就可以通过了，打开之后也是3.83，GPT 4o mini可以使用
